### PR TITLE
[5792] XB (react) - cursor not changing to pencil when hovering xb zones, padding mode not applying styles

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5792

### CODE REVIEW

File: ui/guest/src/react/Field.tsx [hunk 1]
```diff
@@ -41,10 +42,18 @@ export const Field = forwardRef<any, FieldProps>(function <P = {}>(props: FieldP
...
+  const contentType = contentController.getCachedContentType(contentTypeId);
+  const field = contentType?.fields[fieldId];
+
   const passDownProps = {
     ...other,
     ...ice,
     ...componentProps,
+    'data-craftercms-model-id': model.craftercms.id,
+    ...(field &&
+      ['node-selector', 'checkbox-group', 'repeat'].includes(field.type) && { 'data-craftercms-type': 'collection' }),
```
Comment (rart) [inline ui/guest/src/react/Field.tsx]: Why would the checkbox group require the attribute?
```suggestion
      ['node-selector', 'repeat'].includes(field.type) && { 'data-craftercms-type': 'collection' }),
```

Move this whole expression out of the primitive initialization. We don't need to inline everything here.

File: ui/guest/src/react/Field.tsx [hunk 2]
```diff
@@ -41,10 +42,18 @@ export const Field = forwardRef<any, FieldProps>(function <P = {}>(props: FieldP
     ref
   });
   const Component = component as ComponentType<P>;
+
+  const contentTypeId = model.craftercms.contentTypeId;
+  const contentType = contentController.getCachedContentType(contentTypeId);
+  const field = contentType?.fields[fieldId];
+
```
Comment (rart) [inline ui/guest/src/react/Field.tsx]:
```suggestion
  const contentTypeId = model.craftercms.contentTypeId;
  const contentType = contentController.getCachedContentType(contentTypeId);
  const field = contentType?.fields[fieldId];
```

File: ui/guest/src/react/Field.tsx [hunk 3]
```diff
@@ -41,10 +42,18 @@ export const Field = forwardRef<any, FieldProps>(function <P = {}>(props: FieldP
...
   const passDownProps = {
     ...other,
     ...ice,
     ...componentProps,
+    'data-craftercms-model-id': model.craftercms.id,
```
Comment (rart) [inline ui/guest/src/react/Field.tsx]: Use a different attribute (e.g. `data-craftercms-field`). Value is unnecessary. Add comment stating why we're adding these attributes.

File: ui/guest/src/react/RenderField.tsx [hunk 4]
```diff
@@ -61,6 +62,18 @@ export const RenderField = forwardRef<any, RenderFieldProps<{}>>(function <P = {
...
+  const contentTypeId = model.craftercms.contentTypeId;
+  const contentType = contentController.getCachedContentType(contentTypeId);
+  const field = contentType?.fields[fieldId];
+
+  if (field && ['node-selector', 'checkbox-group', 'repeat'].includes(field.type)) {
+    passDownProps['data-craftercms-type'] = 'collection';
+  }
+
+  console.log('contentType', contentType);
+
```
Comment (rart) [inline ui/guest/src/react/RenderField.tsx]:
```suggestion
  // Comment stating why these attributes...
  const contentTypeId = model.craftercms.contentTypeId;
  const contentType = contentController.getCachedContentType(contentTypeId);
  const field = contentType?.fields[fieldId];
  passDownProps['data-craftercms-model-id'] = model.craftercms.id;
  if (field && ['node-selector', 'checkbox-group', 'repeat'].includes(field.type)) {
    passDownProps['data-craftercms-type'] = 'collection';
  }
```

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/guest/src/react/GuestGlobalStyles.tsx [hunk 5]
```diff
@@ -187,7 +187,7 @@ const styles = (
       },
       // region craftercms-ice-on
       [`.${editModeClass}`]: {
-        '[data-craftercms-model-id], & [data-craftercms-model-id] a': {
+        '[data-craftercms-model-id], & [data-craftercms-model-id] a, & [data-craftercms-field]': {
```
Comment (rart) [inline ui/guest/src/react/GuestGlobalStyles.tsx]: missing the internal a style, like the pre-existing style

File: ui/guest/src/react/Field.tsx [hunk 6]
```diff
@@ -19,6 +19,7 @@ import PropTypes from 'prop-types';
 import { ICEProps } from '../models/InContextEditing';
 import ContentInstance from '@craftercms/studio-ui/models/ContentInstance';
 import { useICE } from './hooks';
+import { contentController } from '../index';
```
Comment (rart) [inline ui/guest/src/react/Field.tsx]: import from direct file

File: ui/guest/src/react/RenderField.tsx [hunk 7]
```diff
@@ -20,6 +20,7 @@ import { useICE } from './hooks';
 import { FieldProps, Field } from './Field';
 import { nnou, setProperty } from '@craftercms/studio-ui/utils/object';
 import { extractCollectionItem, value as getModelValue } from '@craftercms/studio-ui/utils/model';
+import { contentController } from '../index';
```
Comment (rart) [inline ui/guest/src/react/RenderField.tsx]: import from direct file

**Review by rart — APPROVED** [review]

**Review by rart — CHANGES_REQUESTED** [review]
review above

File: ui/guest/src/react/RenderField.tsx [hunk 8]
```diff
@@ -61,6 +62,18 @@ export const RenderField = forwardRef<any, RenderFieldProps<{}>>(function <P = {
...
   });
+
+  // `data-craftercms-field` attribute is added to all fields for the elements to get the XB on hover cursor styles.
+  // `data-craftercms-type="collection"` attribute is added to node-selector and repeat fields for the elements to get
+  // the XB padding mode styles.
+  const contentTypeId = model.craftercms.contentTypeId;
+  const contentType = contentController.getCachedContentType(contentTypeId);
+  const field = contentType?.fields[fieldId];
+  passDownProps['data-craftercms-field'] = '';
+  if (field && ['node-selector', 'checkbox-group', 'repeat'].includes(field.type)) {
```
Comment (rart) [inline ui/guest/src/react/RenderField.tsx]: checkbox-group still

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/guest/src/react/Field.tsx [hunk 9]
```diff
@@ -19,6 +19,7 @@ import PropTypes from 'prop-types';
 import { ICEProps } from '../models/InContextEditing';
 import ContentInstance from '@craftercms/studio-ui/models/ContentInstance';
 import { useICE } from './hooks';
+import * as contentController from '../contentController';
```
Comment (rart) [inline ui/guest/src/react/Field.tsx]: import the thing you need

File: ui/guest/src/react/RenderField.tsx [hunk 10]
```diff
@@ -20,6 +20,7 @@ import { useICE } from './hooks';
 import { FieldProps, Field } from './Field';
 import { nnou, setProperty } from '@craftercms/studio-ui/utils/object';
 import { extractCollectionItem, value as getModelValue } from '@craftercms/studio-ui/utils/model';
+import * as contentController from '../contentController';
```
Comment (rart) [inline ui/guest/src/react/RenderField.tsx]: import the thing you need only

**Review by rart — CHANGES_REQUESTED** [review]
Check if we also need Model, RenderRepeat and RenderComponents to have the attributes

**Review by rart — CHANGES_REQUESTED** [review]
Check if we also need Model, RenderRepeat and RenderComponents to have the attributes

**Comment by jvega190** [comment]
Those components don't need the attributes, since `Model` uses `Field`, `RenderRepeat` uses `RenderField` and `RenderComponent` uses `RenderRepeat`. Both `Field` and `RenderField` have all they need to include the attributes (no extra changes needed on the mentioned components)

**Review by rart — APPROVED** [review]

**Pull request merged**